pilot/pkg/proxy/envoy/v1/discovery_test.go

```diff
@@ -949,20 +978,32 @@ func TestListenerDiscoveryRouterError(t *testing.T) {
 	}
 }
 
-func TestListenerDiscoveryRouter(t *testing.T) {
+func TestListenerDiscoveryGateway(t *testing.T) {
```

Comment:
I see some instances of Gateway and some of Router.. Can you please make them consistent? Either Gateway or Router

Reply:
I believe the only bit I haven't changed to `Gateway` proper is the `Type` because it would involve a whole bunch of renaming. I'd be glad to figure that out after this gets merged.